versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/SetMojo.java

```diff
@@ -489,7 +497,7 @@ private void applyChange(
             getLog().debug("Looking for modules which use "
                     + ArtifactUtils.versionlessKey(sourceGroupId, sourceArtifactId) + " as their parent");
 
-            for (Map.Entry<File, Model> stringModelEntry : processAllModules
+            for (Map.Entry<File, Model> stringModelEntry : fullTreeMatch
```

Comment:
If we have `fullTreeMatch` set to `true` we will iterate again by `reactor.entrySet()` we have loop in loop 😄 
see line: 470